webui/react/src/pages/FlatRuns/FlatRunActionButton.tsx

```diff
@@ -0,0 +1,212 @@
+import Button from 'hew/Button';
+import Dropdown, { MenuItem } from 'hew/Dropdown';
+import Icon, { IconName } from 'hew/Icon';
+import { useModal } from 'hew/Modal';
+import { useToast } from 'hew/Toast';
+import { useCallback, useMemo, useState } from 'react';
+
+import BatchActionConfirmModalComponent from 'components/BatchActionConfirmModal';
+import { FilterFormSetWithoutId } from 'components/FilterForm/components/type';
+import usePermissions from 'hooks/usePermissions';
+import FlatRunMoveModalComponent from 'pages/FlatRuns/FlatRunMoveModal';
+import { archiveRuns, deleteRuns, killRuns, unarchiveRuns } from 'services/api';
+import { BulkActionResult, ExperimentAction, FlatRun } from 'types';
+import handleError from 'utils/error';
+import { canActionFlatRun, getActionsForFlatRunsUnion } from 'utils/flatRun';
+import { capitalizeWord } from 'utils/string';
+
+const BATCH_ACTIONS = [
+  ExperimentAction.Move,
+  ExperimentAction.Archive,
+  ExperimentAction.Unarchive,
+  ExperimentAction.Delete,
+  ExperimentAction.Kill,
+] as const;
+
+type BatchAction = (typeof BATCH_ACTIONS)[number];
+
+const ACTION_ICONS: Record<BatchAction, IconName> = {
+  [ExperimentAction.Archive]: 'archive',
+  [ExperimentAction.Unarchive]: 'document',
+  [ExperimentAction.Move]: 'workspaces',
+  [ExperimentAction.Kill]: 'cancelled',
+  [ExperimentAction.Delete]: 'error',
+} as const;
+
+const LABEL_PLURAL = 'runs';
+
+interface Props {
+  isMobile: boolean;
+  selectedRuns: ReadonlyArray<Readonly<FlatRun>>;
+  projectId: number;
+  workspaceId: number;
+  filterFormSetWithoutId: FilterFormSetWithoutId;
+  onActionSuccess?: (action: BatchAction, successfulIds: number[]) => void;
+  onActionComplete?: () => Promise<void>;
```

Comment:
nit: `() => void`

Reply:
probably ` () => void | Promise<void>` is more suitable?  `() => void` works for promise, but editor thinks `await` is not needed if `() => void` is used. i think its misleading. what do you think?

Reply:
sure that sounds good

Reply:
fixed